doc/source/whatsnew/v0.23.2.txt

```diff
@@ -60,6 +60,7 @@ Bug Fixes
 
 - Bug in :meth:`Index.get_indexer_non_unique` with categorical key (:issue:`21448`)
 - Bug in comparison operations for :class:`MultiIndex` where error was raised on equality / inequality comparison involving a MultiIndex with ``nlevels == 1`` (:issue:`21149`)
+- Bug in calculation of group index causing "maximum recursion depth exceeded" errors during ``DataFrame.duplicated`` calls  (:issue:`21524`).
```

Comment:
just say:

bug in ``:func:`DataFrame.duplicated` `` with a large number of columns causing a 'maximum .....'

Reply:
Thanks, I'll update this